src/Recipe.js

```diff
@@ -0,0 +1,58 @@
+const { ingredientsData } = require('../data/ingredients');
+class Recipe {
+  constructor(recipe) {
+    this.id = recipe.id;
+    this.image = recipe.image;
+    this.ingredients = this.createIngredients(recipe.ingredients);
+    this.instructions = recipe.instructions;
+    this.name = recipe.name;
+    this.tags = recipe.tags;
+  }
+
+  createIngredients(ingredients) {
+    const fancyIngredients = ingredients.map(ingredient => {
+      const fancyIngredient = {
+        id: ingredient.id,
+        amount: ingredient.quantity.amount,
+        unit: ingredient.quantity.unit,
+      };
+      fancyIngredient.name = this.getIngredientData(
+        fancyIngredient.id, 
+        'name'
+      ); 
+      fancyIngredient.costPerUnit = this.getIngredientData(
+        fancyIngredient.id, 
+        'estimatedCostInCents'
+      );
+      fancyIngredient.totalCostInCents = 
+        fancyIngredient.amount * fancyIngredient.costPerUnit;
+
+      return fancyIngredient;
+    });
+
+    return fancyIngredients;
```

Comment:
I think we'll need a different variable name. What does 'fancy' mean??

Reply:
Hahaha I agree but couldn't think of anything better. What it means here is "ingredient object that has more data in it than just the few properties that came in with the initial recipe object". In other words, recipe data arrives with ingredients that have an id and a quantity (amount and unit). But we also need the ingredient names and prices, and those have to be looked up in the `ingredients.js` data file. So a "fancy" ingredient object is one that has *all* the ingredient data we'll need while working with and displaying recipes.